superset-frontend/src/SqlLab/components/SqlEditor/index.jsx

```diff
@@ -229,13 +229,20 @@ const SqlEditor = ({
     }
   };
 
-  useState(() => {
+  const runQuery = () => {
+    if (database) {
+      startQuery();
+    }
+  };
+
+  // hack the useMemo hook to imitate componentWillMount
+  useMemo(() => {
```

Comment:
I think the  `useEffect` hook be more appropriate here and not be considered a hack, but recommended practice from React.

When using a `
`
useEffect() => {
}, [])
`
With empty array it tells the functional component to only run the effect one time.

Notes section of: https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects

"If you want to run an effect and clean it up only once (on mount and unmount), you can pass an empty array ([]) as a second argument. This tells React that your effect doesn’t depend on any values from props or state, so it never needs to re-run. This isn’t handled as a special case — it follows directly from how the dependencies array always works." 

Basically same exact code just swapping out the `useMemo` with `useEffect` to align with React's best practices and remove the hack comment

Reply:
@eric-briscoe am I correct
useMemo(() => {}, []); is componentWillMount event.
useEffect(() => {}, []); is componentDidMount

Reply:
@EugeneTorap `useEffect` is intended to replace all of the Class lifecycle methods so with functional components React is saying you no longer would try to differentiate between componentDidMount, componentDidUpdate, and componentWillUnmount, just use `useEffect`.

`Tip If you’re familiar with React class lifecycle methods, you can think of useEffect Hook as componentDidMount, componentDidUpdate, and componentWillUnmount combined.` https://reactjs.org/docs/hooks-effect.html 

useMemo is for a very specific purpose and even React recommends not using it until you have a working component and start experience performance issues. useMemo runs during render and is not intended to produce side effects.

`Remember that the function passed to useMemo runs during rendering. Don’t do anything there that you wouldn’t normally do while rendering. For example, side effects belong in useEffect, not useMemo.` https://reactjs.org/docs/hooks-reference.html#usememo